src/searchbar.cpp

```diff
@@ -177,6 +177,21 @@ void SearchBarLineEdit::fetchMoreSuggestions()
 
 void SearchBarLineEdit::onScroll(int value)
 {
+    if (!m_completer.popup()->currentIndex().isValid())
+    {
+        m_aboutToScrollPastEnd = false;
+
+        /* m_completer.popup()->currentIndex() being invalid means the list
+           has been scrolled from bottom to top. We undo this here, as it avoids
+           scroll bar flicker as well. Block signal to avoid recursion.
+        */
+        auto old = m_completer.popup()->verticalScrollBar()->blockSignals(true);
+        m_completer.popup()->scrollToBottom();
+        m_completer.popup()->verticalScrollBar()->blockSignals(old);
+        
+        return fetchMoreSuggestions();
```

Comment:
What about packaging this piece of code (without the comment which should stay here) into a new helper method `SearchBarLineEdit::onScrollPastEnd()` and reusing it for the other call of `fetchMoreSuggestions()`?

Reply:
Done